src/Compilers/Core/Portable/PEWriter/MetadataWriter.cs

```diff
@@ -2321,6 +2321,8 @@ private void PopulateFieldRvaTableRows(BlobBuilder mappedFieldDataWriter)
                 }
 
                 int offset = mappedFieldDataWriter.Count;
+                Debug.Assert(offset % ManagedPEBuilder.MappedFieldDataAlignment == 0, "Expected last write to end at alignment boundary");
```

Comment:
nit: consider asserting with specific value (8)

Reply:
Using the named const seems better to me.  But since you asked, I added an assert that the named const is 8 as well.